### PR TITLE
Add Twilio and Sendgrid providers

### PR DESCRIPTION
**Pull request by Aschen**
## Description

This plugins allows to send messages by using different message providers.

The available providers are Twilio for SMS and Sendgrid for emails.

It's possible to register multiple accounts for each provider. Each account is a set of credentials used to send message.
In this way each tenant can register his own Twilio or Sendgrid account.

https://next-docs.kuzzle.io/official-plugins/hermes-messenger/1/

### CODE REVIEW

File: .github/workflows/push_master.workflow.yml [hunk 1]
```diff
@@ -48,6 +48,36 @@ jobs:
...
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+      - uses: actions/setup-node@v1
+        with:
+          node-version: '12.x'
+          registry-url: 'https://registry.npmjs.org'
+      - run: npm install
```
Comment (Leodau) [inline .github/workflows/push_master.workflow.yml:76]: Is `npm install` the good option here, or `npm ci` could be better?

File: doc/1/controllers/sendgrid/sendEmail/index.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
+---
+type: page
+code: true
+title: sendEmail
+description: Send an email with a Sendgrid account
+---
+
+# sendEmail
+
+Sends an email using one of the registered Sendgrid account.
```
Comment (Leodau) [inline doc/1/controllers/sendgrid/sendEmail/index.md:10]:
```suggestion
Sends an email using one of the registered Sendgrid accounts.
```

File: doc/1/controllers/twilio/sendSms/index.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
+---
+type: page
+code: true
+title: sendSms
+description: Send a SMS with a Twilio account
+---
+
+# sendSms
+
+Sends a SMS using one of the registered Twilio account.
```
Comment (Leodau) [inline doc/1/controllers/twilio/sendSms/index.md:10]:
```suggestion
Sends an SMS using one of the registered Twilio account.
```

File: doc/1/controllers/twilio/sendSms/index.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
+---
+type: page
+code: true
+title: sendSms
+description: Send a SMS with a Twilio account
```
Comment (Leodau) [inline doc/1/controllers/twilio/sendSms/index.md:5]:
```suggestion
description: Send an SMS with a Twilio account
```

File: README.md [hunk 5]
```diff
@@ -1,5 +1,66 @@
 # Hermes Messenger 
 
+This Kuzzle plugin allows to send messages using various external providers.
```
Comment (Leodau) [inline README.md:3]:
```suggestion
This Kuzzle plugin allows you to send messages using various external providers.
```
or
```suggestion
This Kuzzle plugin gives the application the ability to send messages using various external providers.
```

File: doc/1/guides/overview/index.md [hunk 6]
```diff
@@ -6,3 +6,42 @@ description: Overview of the Hermes Messenger plugin features and main concepts
 order: 100
 ---
 
+# Hermes Messenger
+
+This plugin allows to send various types of messages (SMS, email) using external providers.
```
Comment (Leodau) [inline doc/1/guides/overview/index.md:11]:
```suggestion
This plugin allows you to send various types of messages (SMS, email) using external providers.
```
or
```suggestion
This plugin gives the application the ability to send various types of messages (SMS, email) using external providers.
```

File: doc/1/guides/overview/index.md [hunk 7]
```diff
@@ -6,3 +6,42 @@ description: Overview of the Hermes Messenger plugin features and main concepts
...
+
+Available providers SMS providers:
+  - Twilio
+
+Available email providers:
+  - Sendgrid
+
+## Accounts Management
+
+It's possible to register differents accounts for each providers. Then API actions used to send messages can be used with different accounts.
```
Comment (Leodau) [inline doc/1/guides/overview/index.md:21]:
```suggestion
It is possible to register differents accounts for each provider. Then API actions used to send messages can be used with different accounts.
```

File: package.json [hunk 8]
```diff
@@ -30,6 +30,7 @@
     "twilio": "^3.62.0"
   },
   "devDependencies": {
+    "express": "^4.17.1",
```
Comment (Leodau) [inline package.json:33]: :eyes:

**Review by Leodau — APPROVED** [review]
Awesome!

File: package.json (hunk 8 above)
Comment (Aschen) [reply package.json:33]: Yes it was needed by Twilio in dev mod, I'm not sure why :facepalm:

**Pull request merged**